tests/framework/extensions/prime/primechecks.go

```diff
@@ -0,0 +1,92 @@
+package primechecks
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	client "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	PodResourceSteveType = "pod"
+	rancherImage         = "rancher"
+)
+
+// CheckUIBrand checks the UI brand of Rancher Prime. If the Rancher instance is not Rancher Prime, the UI brand should be blank.
+func CheckUIBrand(client *rancher.Client, isPrime bool, brand *client.Setting, rancherBrand string) (string, error) {
+	if isPrime {
+		if rancherBrand == brand.Value {
+			logrus.Infof("Rancher Prime UI brand: %s", brand.Value)
+			return brand.Value, nil
+		}
+	}
+
+	return brand.Value, nil
+}
+
+// CheckVersion checks the if Rancher Prime is set to true and the version of Rancher.
+func CheckVersion(isPrime bool, rancherVersion string, serverConfig *rancherversion.Config) (string, error) {
+	if isPrime {
+		if rancherVersion == serverConfig.RancherVersion {
+			logrus.Infof("Rancher Prime: %t | Version: %s", isPrime, serverConfig.RancherVersion)
+			return serverConfig.RancherVersion, nil
+		}
+	}
+
+	return serverConfig.RancherVersion, nil
+}
+
+// CheckSystemDefaultRegistry checks if the system default registry is set to the expected value.
+func CheckSystemDefaultRegistry(isPrime bool, primeRegistry string, registry *client.Setting) (string, error) {
+	if isPrime {
+		if primeRegistry == registry.Value {
+			logrus.Infof("Rancher Prime system default registry: %s", registry.Value)
```

Comment:
Same here

Reply:
Removed the error as a return value and just kept it as a string being returned.